test/run-spec-wasm2c.py

```diff
@@ -509,7 +509,7 @@ def main(args):
             '--enable-multi-memory': options.enable_multi_memory})
 
         options.cflags += shlex.split(os.environ.get('WASM2C_CFLAGS', ''))
-        cc = utils.Executable(options.cc, *options.cflags, forward_stderr=True,
+        cc = utils.Executable(options.cc, after_args=options.cflags, forward_stderr=True,
```

Comment:
We want the environment-specified CFLAGS to go to the end of the argument list so that CMake can supply something like WASM2C_CFLAGS="-g -O0" (on a Debug build) and have that override the default of `-O2` in run-spec-wasm2c.py (because it comes later in the compiler invocation).

Another option would be for run-spec-wasm2c.py to check for the existence of the WASM2C_CFLAGS environment variable, and only use '-O2' if that env var is undefined? But I think it's probably still helpful to put the environment variable flags at the end of the argument list because people might want to do other things (e.g. turn off warnings, etc.) so you want the user-specified flags to override the defaults.

Reply:
Split out into #2134.